### ISSUE TITLE
0.12.0 AsDocComment casting error

### ISSUE THREAD

**Issue by Harbs** [BowlerHatLLC/vscode-nextgenas#246]
I'm currently getting many errors like this in v. 0.12.0:
````
[Error - 15:14:40] Request textDocument/hover failed.
  Message: Internal error.
  Code: -32603 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:63)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:116)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:257)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:186)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:188)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:90)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:95)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:61)
	... 11 more
Caused by: java.lang.ClassCastException: com.nextgenactionscript.vscode.asdoc.VSCodeASDocComment cannot be cast to org.apache.royale.compiler.asdoc.royale.ASDocComment
	at org.apache.royale.compiler.internal.projects.RoyaleJSProject.addDependency(RoyaleJSProject.java:116)
	at org.apache.royale.compiler.internal.scopes.ASProjectScope.findDefinitionByName(ASProjectScope.java:1591)
	at org.apache.royale.compiler.internal.scopes.ASProjectScope.getPropertyForScopeChain(ASProjectScope.java:1568)
	at org.apache.royale.compiler.internal.scopes.ASScope.findProperty(ASScope.java:1154)
	at org.apache.royale.compiler.internal.scopes.ASScope.findProperty(ASScope.java:1061)
	at org.apache.royale.compiler.internal.scopes.ASScope.findProperty(ASScope.java:1277)
	at org.apache.royale.compiler.internal.scopes.ASScope.findProperty(ASScope.java:999)
	at org.apache.royale.compiler.internal.scopes.ASScopeCache.findProperty(ASScopeCache.java:153)
	at org.apache.royale.compiler.internal.scopes.ASScope.findProperty(ASScope.java:1428)
	at org.apache.royale.compiler.internal.definitions.references.LexicalReference.resolve(LexicalReference.java:72)
	at org.apache.royale.compiler.internal.definitions.DefinitionBase.resolveType(DefinitionBase.java:1061)
	at org.apache.royale.compiler.internal.definitions.DefinitionBase.resolveType(DefinitionBase.java:1101)
	at org.apache.royale.compiler.internal.definitions.ClassDefinition.resolveBaseClass(ClassDefinition.java:362)
	at org.apache.royale.compiler.internal.definitions.ClassDefinitionBase$ClassIterator.next(ClassDefinitionBase.java:523)
	at org.apache.royale.compiler.internal.definitions.ClassDefinitionBase$ClassIterator.next(ClassDefinitionBase.java:489)
	at org.apache.royale.compiler.internal.scopes.TypeScope.addImplicitOpenNamespaces(TypeScope.java:557)
	at org.apache.royale.compiler.internal.scopes.ScopeView.addImplicitOpenNamespaces(ScopeView.java:115)
	at org.apache.royale.compiler.internal.scopes.ASScope.getNamespaceSetImpl(ASScope.java:558)
	at org.apache.royale.compiler.internal.scopes.ASScopeCache.getNamespaceSet(ASScopeCache.java:395)
	at org.apache.royale.compiler.internal.scopes.ASScope.getNamespaceSet(ASScope.java:515)
	at org.apache.royale.compiler.internal.scopes.ASScope.addNamespacesFromContainingScope(ASScope.java:578)
	at org.apache.royale.compiler.internal.scopes.ASScope.getNamespaceSetImpl(ASScope.java:561)
	at org.apache.royale.compiler.internal.scopes.ASScopeCache.getNamespaceSet(ASScopeCache.java:395)
	at org.apache.royale.compiler.internal.scopes.ASScope.getNamespaceSet(ASScope.java:515)
	at org.apache.royale.compiler.internal.scopes.ASScope.getNamespaceSetForNameImpl(ASScope.java:484)
	at org.apache.royale.compiler.internal.scopes.ASScopeCache.getNamespaceSetForName(ASScopeCache.java:557)
	at org.apache.royale.compiler.internal.scopes.ASScope.getNamespaceSetForName(ASScope.java:447)
	at org.apache.royale.compiler.internal.scopes.ASScopeCache.findProperty(ASScopeCache.java:151)
	at org.apache.royale.compiler.internal.scopes.ASScope.findProperty(ASScope.java:1428)
	at org.apache.royale.compiler.internal.scopes.ASScope.findProperty(ASScope.java:1423)
	at org.apache.royale.compiler.internal.tree.as.IdentifierNode.resolve(IdentifierNode.java:394)
	at org.apache.royale.compiler.internal.tree.as.IdentifierNode.resolveType(IdentifierNode.java:466)
	at org.apache.royale.compiler.internal.tree.as.IdentifierNode.resolveMemberRef(IdentifierNode.java:885)
	at org.apache.royale.compiler.internal.tree.as.IdentifierNode.resolve(IdentifierNode.java:388)
	at com.nextgenactionscript.vscode.ActionScriptTextDocumentService.actionScriptHoverWithNode(ActionScriptTextDocumentService.java:2299)
	at com.nextgenactionscript.vscode.ActionScriptTextDocumentService.actionScriptHover(ActionScriptTextDocumentService.java:2280)
	at com.nextgenactionscript.vscode.ActionScriptTextDocumentService.hover(ActionScriptTextDocumentService.java:474)
	... 16 more
````

**Comment by joshtynjala** [comment]
Commit 50310050f62d741d3ce4a4aab06e07eb9c2610bf in royale-compiler causes this issue. It needs to use the IAsDocComment interface instead of the ASDocComment class.

**Comment by Harbs** [comment]
I believe I fixed this with apache/royale-compiler@9dbcddeb9aa678ce9331aa305dc77c1544c15d0d

**Comment by joshtynjala** [comment]
Great! Thanks
